x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealmTests.java

```diff
@@ -108,6 +104,122 @@ public void shutdown() throws Exception {
         terminate(this.threadPool);
     }
 
+    /**
+     * Generate a static JWT realm config, and request headers, which the realm can successfully authenticate.
+     *
+     * Print these artifacts, so they can be imported into this integration test project for verifying loading of JWT realm config:
+     *  - x-pack/plugin/security/qa/smoke-test-all-realms/build.gradle
+     *
+     * If something needs tweaking, change the code here and re-run it. The new artifacts can be used to update that smoke test project.
+     * @throws Exception Error if something goes wrong.
+     */
+    public void testCreateJwtSmokeTestRealm() throws Exception {
+        // static HMAC key (password UTF-8 bytes, for OIDC HMAC Key encoding compatibility)
+        final byte[] hmacKeyOidcBytes = "hmac-oidc-key-string-for-hs256-algorithm".getBytes(StandardCharsets.UTF_8);
+        final OctetSequenceKey hmacJwkOidc = new OctetSequenceKey.Builder(hmacKeyOidcBytes).build();
+
```

Comment:
This method is not a test method. I think we should move it into its own helper file so it does not run with test suite but can be executed manually if needed to.

Reply:
I did think about moving it to another class, but it has dependencies on helper methods in JwtRealmTests, so I left it in there.

Also, it is actually doing authentication and authorization tests. The order of steps are:
1. Create the issuer, realm, user, and JWT
2. Run the authc/authz authentication test helper to verify everything works together
3. Print all of the artifacts

Printing the realm settings and request header are only possible if it passes the authc/authz tests, so it is actually a test.

Reply:
Do you still think it should be moved to a different test class? If yes, I will need to re-run it and put the changed output in the other PR.

Reply:
Created new JwtRealmTestCase, and split public test methods between two classes. Resolved.